tox.ini

```diff
@@ -1,22 +1,47 @@
 [tox]
-minversion = 3.5.1
+minversion = 3.8.0
 skip_missing_interpreters = true
-envlist = py35,
-          py36,
-          py37,
-          lint,
-          type,
-          docs
+envlist =
+    py35,
+    py36,
+    py37,
+    coverage,
+    lint,
+    type,
+    docs,
 isolated_build = true
 
 [testenv]
 description = run the test driver with {basepython}
+setenv = COVERAGE_FILE={toxworkdir}/.coverage.{envname}
 passenv = PYTEST_XDIST_WORKER_COUNT NEWSEMANAL
 deps = -rtest-requirements.txt
-commands = pytest {posargs}
+commands =
+    python -m pytest {posargs} \
+    --cov mypy \
+    --cov-config setup.cfg
```

Comment:
Does this incur a performance penalty? If so, I don't think I want it as the default

Reply:
Decided to profile it:

With:
      630.19 real      2068.13 user       129.78 sys
Without:
     307.89 real       942.10 user       108.35 sys